subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -530,19 +541,79 @@ static int cell_pos_cb_send(const char *const rx_buf)
 	return -EFTYPE;
 }
 
+void agps_process(const char * const buf, const size_t buf_len)
+{
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+	int ret = nrf_cloud_agps_process(buf, buf_len);
+
+	LOG_DBG("A-GPS data processed; result: %d", ret);
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	if (ret == 0) {
+		ret = nrf_cloud_pgps_notify_prediction();
```

Comment:
Remind me again why this must be called in response to agps data receipt? I don't question it, but I would like to fully understand

Reply:
i don't know, but i bet @plskeggs does

Reply:
If both A-GPS and P-GPS are enabled, everything but ephemerides and almanacs are handled by A-GPS.  In this configuration, it is important to check, after receiving A-GPS data, whether any further assistance is needed by the modem for ephemerides, which would come from P-GPS (usually, in stored predictions in flash).

Reply:
maybe that should be a comment in the code somewhere?  or maybe it already is?